### PR TITLE
Saml Second Factor Handling

### PR DESCRIPTION
**Pull request by DRvanR**
- [x] Add tests for event / command

### CODE REVIEW

**Comment by DRvanR** [comment]
pfuh

**Comment by DRvanR** [comment]
@rjkip

File: src/Surfnet/Stepup/Identity/Value/Institution.php [hunk 1]
```diff
@@ -33,11 +33,11 @@ class Institution implements JsonSerializable
      */
     public function __construct($institution)
     {
-        if (!is_string($institution) || $institution === '') {
+        if (!is_string($institution) || strlen(trim($institution)) === 0) {
```
Comment (rjkip) [inline src/Surfnet/Stepup/Identity/Value/Institution.php]: Just out of curiosity: does `strlen() === 0` have some advantage over `=== ''`?

File: src/Surfnet/Stepup/Identity/Event/GssfPossessionProvenEvent.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    {
+        return [
+            'identity_id'               => (string)$this->identityId,
+            'second_factor_id'          => (string)$this->secondFactorId,
+            'stepup_provider'           => (string)$this->stepupProvider,
+            'gssf_id'                   => (string)$this->gssfId,
+            'email_verification_window' => $this->emailVerificationWindow->serialize(),
+            'email_verification_nonce'  => (string)$this->emailVerificationNonce,
+            'common_name'               => (string)$this->commonName,
+            'email'                     => (string)$this->email,
```
Comment (rjkip) [inline src/Surfnet/Stepup/Identity/Event/GssfPossessionProvenEvent.php]: We separate the cast from its subject with a space throughout the project.

File: src/Surfnet/Stepup/Identity/Value/Institution.php (hunk 1 above)
Comment (DRvanR) [inline src/Surfnet/Stepup/Identity/Value/Institution.php]: bar the also added `trim`, nope :wink:

File: src/Surfnet/Stepup/Identity/Event/GssfPossessionProvenEvent.php (hunk 2 above)
Comment (DRvanR) [inline src/Surfnet/Stepup/Identity/Event/GssfPossessionProvenEvent.php]: silly `ctrl+alt+l` reformatting that -.-

File: src/Surfnet/Stepup/Tests/Identity/Value/InstitutionTest.php [hunk 3]
```diff
@@ -27,8 +27,9 @@ class InstitutionTest extends UnitTest
      * @test
      * @group domain
      * @dataProvider invalidValueProvider
-     *
      * @expectedException \Surfnet\Stepup\Exception\InvalidArgumentException
+     *
+     * @parma mixed $invalidValue
```
Comment (rjkip) [inline src/Surfnet/Stepup/Tests/Identity/Value/InstitutionTest.php]: image

File: src/Surfnet/Stepup/Tests/Identity/Value/InstitutionTest.php [hunk 4]
```diff
@@ -46,7 +47,7 @@ public function two_institutions_with_the_same_value_are_equal()
         $different = new Institution('A');
 
         $this->assertTrue($institution->equals($theSame));
-        $this->assertFalse($institution->equals($different));;
+        $this->assertFalse($institution->equals($different));
```
Comment (rjkip) [inline src/Surfnet/Stepup/Tests/Identity/Value/InstitutionTest.php]: Would be nice to test `'a' === '   a '`.

**Comment by DRvanR** [comment]
@rjkip All casts have been fixed

**Comment by rjkip** [comment]
:+1:

**Pull request merged**